src/Compiler/OnnxMlirCompiler.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "OnnxMlirCompiler.h"
+#include "CompilerUtils.hpp"
+
+extern "C" {
+namespace onnx_mlir {
+ONNX_MLIR_EXPORT int CompileFromFile(const char *inputFilename,
+    const char *outputBaseName, EmissionTargetType emissionTarget,
+    const char *mcpu, const char *mtriple) {
+  mlir::OwningModuleRef module;
+  mlir::MLIRContext context;
+  std::string mcpuString;
```

Comment:
Suggestion: could we have the code that is in common between the two CompileFromFile and CompileFromArray in a common function, say SetCompileContext, handling them cpu, mtriple, dialect registration?